### PR TITLE
fix: Allow fromEnv instantiation from oauth client

### PR DESCRIPTION
**Pull request by Gara64**
#447 could not work with the [cozy-konnector-dev](https://github.com/konnectors/libs/tree/master/packages/cozy-jobs-cli) script because it uses an OAuth authentication and `fromEnv` was expecting a string from `COZY_CREDENTIALS`
This PR fixes this behaviour.

### CODE REVIEW

File: packages/cozy-client/src/CozyClient.js [hunk 1]
```diff
@@ -103,9 +103,19 @@ class CozyClient {
         'Env used to instantiate CozyClient must have COZY_URL and COZY_CREDENTIALS'
       )
     }
+    let oauth
+    let token
+    try {
+      oauth = JSON.parse(COZY_CREDENTIALS)
+      token = JSON.parse(COZY_CREDENTIALS)
```
Comment (ptbrowne) [inline packages/cozy-client/src/CozyClient.js]: Why do we have both ?

If parsing is successful, both oauth and token contain the same value, and I do not think the result is valable for the `token` option. Maybe you can remove the `token = JSON.parse` line ?

Can you add a test testing this behavior ?

Comment (edas) [reply packages/cozy-client/src/CozyClient.js]: The logic should probably be from https://github.com/konnectors/libs/blob/db23f283acb10c2ed1651e07855495fe5396f583/packages/cozy-konnector-libs/src/libs/cozyclient.js#L14

File: packages/cozy-client/src/CozyClient.js [hunk 2]
```diff
@@ -103,9 +103,19 @@ class CozyClient {
...
+    } catch (e) {
+      token = COZY_CREDENTIALS.trim()
+    }
+
     return new CozyClient({
+      oauth: oauth,
```
Comment (edas) [inline packages/cozy-client/src/CozyClient.js]: https://github.com/konnectors/libs/blob/db23f283acb10c2ed1651e07855495fe5396f583/packages/cozy-konnector-libs/src/libs/cozyclient.js#L54

File: packages/cozy-client/src/CozyClient.spec.js [hunk 3]
```diff
@@ -84,6 +84,23 @@ describe('CozyClient initialization', () => {
...
+        accessToken: token
+      }
+    }
+    global.process.env.COZY_URL = url
+    global.process.env.COZY_CREDENTIALS = JSON.stringify(creds)
+    global.process.env.NODE_ENV = 'development'
+    client = CozyClient.fromEnv()
+    expect(client.stackClient.uri).toBe(url)
+    expect(client.stackClient.token.accessToken).toBe(token)
+    global.process.env.NODE_ENV = 'test'
```
Comment (ptbrowne) [inline packages/cozy-client/src/CozyClient.spec.js]: It is a bit unhygienic to let COZY_URL etc.. inside process.env after the test. Additionally, if one of the expect fails, the process.env.NODE_ENV is not set back to test.

I think you should use the second parameter env of fromEnv to mock the environment. 

```
client = CozyClient.fromEnv({ COZY_URL: 'blabla', COZY_CREDENTIALS: 'balbla' , NODE_ENV: 'bblabla' })
```

**Comment by ptbrowne** [comment]
Thanks :tada: LGTM, you can merge :)

**Comment by ptbrowne** [comment]
cc @doubleface

**Pull request merged**